registry/storage/driver/gcs/gcs_test.go

```diff
@@ -25,32 +26,51 @@ var skipGCS func() string
 
 func init() {
 	bucket := os.Getenv("REGISTRY_STORAGE_GCS_BUCKET")
-	keyfile := os.Getenv("REGISTRY_STORAGE_GCS_KEYFILE")
 	credentials := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
 
+	// Skip GCS storage driver tests if environment variable parameters are not provided
+	skipGCS = func() string {
+		if bucket == "" || credentials == "" {
+			return "The following environment variables must be set to enable these tests: REGISTRY_STORAGE_GCS_BUCKET, REGISTRY_STORAGE_GCS_CREDS"
+		}
+		return ""
+	}
+
+	if skipGCS() != "" {
+		return
+	}
+
 	root, err := ioutil.TempDir("", "driver-")
 	if err != nil {
 		panic(err)
 	}
 	defer os.Remove(root)
 
-	gcsDriverConstructor = func(rootDirectory string) (storagedriver.StorageDriver, error) {
-
-		parameters := driverParameters{
-			bucket,
-			keyfile,
-			rootDirectory,
+	_, err = os.Stat(credentials)
+	if err == nil {
+		jsonKey, err := ioutil.ReadFile(credentials)
+		if err != nil {
+			panic(fmt.Sprintf("Unable to read credentials from file : %s", err))
 		}
+		credentials = string(jsonKey)
+	}
 
-		return New(parameters)
+	// Assume that the file contents are within the environment variable
```

Comment:
Maybe this comment should be clarified to note that we're assuming the credentials are in the environment variable if the credentials file was not found